R/FilterStatesMAE.R

```diff
@@ -125,96 +131,88 @@ MAEFilterStates <- R6::R6Class( # nolint
     #' @description
     #' Returns active `FilterState` objects.
     #'
-    #' Gets all active filters from this dataset in form of the nested list.
-    #' The output list can be used as input to `self$set_filter_state`.
+    #' Gets all filter state information from this dataset.
+    #'
+    #' @return `teal_slices`
     #'
-    #' @return `list` with elements number equal number of `FilterStates`.
     get_filter_state = function() {
-      lapply(private$state_list_get("y"), function(x) x$get_state())
+      slices <- lapply(private$state_list$y(), function(x) x$get_state())
+      excluded_varnames <- structure(
+        list(setdiff(colnames(SummarizedExperiment::colData(private$data)), private$filterable_varnames)),
+        names = private$dataname)
+      excluded_varnames <- Filter(function(x) !identical(x, character(0)), excluded_varnames)
+
+      do.call(filter_settings, c(slices, list(exclude = excluded_varnames, count_type = private$count_type)))
     },
 
     #' @description
     #' Set filter state
     #'
     #' @param state (`named list`)\cr
-    #'   should contain values which are initial selection in the `FilterState`.
-    #'   Names of the `list` element should correspond to the name of the
-    #'   column in `colData(data)`.
-    #' @return `NULL`
+    #'    `teal_slice` objects should contain the field `target = "y"`
+    #'
+    #' @return `NULL` invisibly
+    #'
     set_filter_state = function(state) {
-      logger::log_trace("MAEFilterState$set_filter_state initializing, dataname: { private$dataname }")
-      checkmate::assert_list(state, null.ok = TRUE, names = "named")
+      checkmate::assert_class(state, "teal_slices")
 
-      data <- private$data
-      data_reactive <- private$data_reactive
-
-      # excluding not supported variables
-      state_varnames <- names(state)
-      filterable_varnames <- get_supported_filter_varnames(SummarizedExperiment::colData(data))
-      excluded_varnames <- setdiff(state_varnames, filterable_varnames)
-      if (length(excluded_varnames) > 0) {
-        excluded_varnames_str <- toString(excluded_varnames)
-        warning(
-          "These columns filters were excluded: ",
-          excluded_varnames_str,
-          " from dataset ",
-          private$dataname
+      logger::log_trace("{ class(self)[1] }$set_filter_state initializing, dataname: { private$dataname }")
+
+      # Drop teal_slices that refer to excluded variables.
+      varnames <- unique(unlist(extract_feat(state, "varname")))
+      filterable <- get_supported_filter_varnames(SummarizedExperiment::colData(private$data))
+      if (!all(varnames %in% filterable)) {
+        excluded_variables <- toString(dQuote(setdiff(varnames, filterable), q = FALSE))
+        state <- extract_fun_s(
+          state,
+          sprintf("!varname %%in%% c(%s)", )
```

Comment:
sprintf is missing argument
```suggestion
          sprintf("!varname %%in%% c(%s)", )
```